docs/startup/preview/index.md

```diff
@@ -0,0 +1,530 @@
+ class OrderShipmentJoiner implements JoinFunction<String, OrderRecord, ShipmentRecord, FulFilledOrderRecord> {
+   @Override
+   public FulFilledOrderRecord apply(OrderRecord message, ShipmentRecord otherMessage) {
+     return new FulFilledOrderRecord(message.orderId, message.orderTimestamp, otherMessage.shipTimestamp);
+   }
+
+   @Override
+   public String getFirstKey(OrderRecord message) {
+     return message.orderId;
+   }
+
+   @Override
+   public String getSecondKey(ShipmentRecord message) {
+     return message.orderId;
+   }
+ }
+
+{% endhighlight %}
+
+
+### Window
+
+#### Windowing Concepts
+**Windows, Triggers, and WindowPanes**: The window operator groups incoming messages in the MessageStream into finite windows. Each emitted result contains one or more messages in the window and is called a WindowPane.
+
+A window can have one or more associated triggers which determine when results from the window are emitted. Triggers can be either [early triggers](/learn/documentation/{{site.version}}/api/javadocs/org/apache/samza/operators/windows/Window.html#setEarlyTrigger-org.apache.samza.operators.triggers.Trigger-) that allow emitting results speculatively before all data for the window has arrived, or late triggers that allow handling late messages for the window.
+
+**Aggregator Function**: By default, the emitted WindowPane will contain all the messages for the window. Instead of retaining all messages, you typically define a more compact data structure for the WindowPane and update it incrementally as new messages arrive, e.g. for keeping a count of messages in the window. To do this, you can provide an aggregating [FoldLeftFunction](/learn/documentation/{{site.version}}/api/javadocs/org/apache/samza/operators/functions/FoldLeftFunction.html) which is invoked for each incoming message added to the window and defines how to update the WindowPane for that message.
+
+**Accumulation Mode**: A window’s accumulation mode determines how results emitted from a window relate to previously emitted results. The accumulation mode can either be discarding or accumulating.
+
+A *discarding window* clears all state for the window at every emission. Each emission will only correspond to new messages that arrived since the previous emission for the window.
+
+An *accumulating window* retains window results from previous emissions. Each emission will contain all messages that arrived since the beginning of the window.
+
+#### Window Types:
+The Samza high-level API currently supports tumbling and session windows.
+
+**Tumbling Window**: A tumbling window defines a series of contiguous, fixed size time intervals in the stream.
+
+Examples:
+{% highlight java %}
+
+// Group the pageView stream into 3 second tumbling windows keyed by the userId.
+MessageStream<PageView> pageViews = ...
+MessageStream<WindowPane<String, Collection<PageView>>> =
```

Comment:
You mean AccumulationModes? Those really only apply for early triggers. I think an example with early triggers and accumulation modes might be to complicated for an inline example. 

@vjagadish1989 do we have any in the cookbook that we can reference?